src/overlays/actors/ovl_Boss_Goma/z_boss_goma.c

```diff
@@ -21,115 +111,1836 @@ const ActorInit Boss_Goma_InitVars = {
+                }
+                Audio_SetBGM(0x1B);
+                gSaveContext.eventChkInf[7] |= 1;
+            }
+            if (func_800A56C8(&this->skelanime, this->currentAnimFrameCount)) {
+                this->actionState = 140;
+                SkelAnime_ChangeAnim(&this->skelanime, &D_06000AE8, 1.0f, 0.0f,
+                                     SkelAnime_GetFrameCount(&D_06000AE8.genericHeader), 0, -10.0f);
+                this->framesUntilNextAction = 20;
+            }
+            break;
+        case 140:
+            SkelAnime_FrameUpdateMatrix(&this->skelanime);
+            Math_SmoothScaleMaxF(&this->subCameraAt.y, this->actor.posRot2.pos.y, 0.1f, 10.0f);
+            // timer set to 20 by previous state
+            if (this->framesUntilNextAction == 0) {
+                this->framesUntilNextAction = 30;
+                this->actionState = 150;
+                Gameplay_ChangeCameraStatus(globalCtx, 0, 3);
+            }
+            break;
+        case 150:
+            SkelAnime_FrameUpdateMatrix(&this->skelanime);
+            Math_SmoothScaleMaxMinF(&this->subCameraEye.x, this->actor.posRot.pos.x + 150.0f, 0.2f, 100.0f, 0.1f);
+            Math_SmoothScaleMaxMinF(&this->subCameraEye.y, this->actor.posRot.pos.y + 20.0f, 0.2f, 100.0f, 0.1f);
+            Math_SmoothScaleMaxMinF(&this->subCameraEye.z, this->actor.posRot.pos.z + 220.0f, 0.2f, 100.0f, 0.1f);
+            // timer set to 30 by previous state
+            if (this->framesUntilNextAction == 0) {
+                cam = Gameplay_GetCamera(globalCtx, 0);
+                cam->eye = this->subCameraEye;
+                cam->eyeNext = this->subCameraEye;
+                cam->at = this->subCameraAt;
+                func_800C08AC(globalCtx, this->subCameraId, 0);
+                this->subCameraId = 0;
+                BossGoma_SetupFloorMain(this);
+                this->disableGameplayLogic = false;
+                this->patienceTimer = 200;
+                func_80064534(globalCtx, &globalCtx->csCtx);
+                func_8002DF54(globalCtx, &this->actor, 7);
+            }
+            break;
+    }
+
+    if (this->subCameraId != 0) {
+        func_800C04D8(globalCtx, this->subCameraId, &this->subCameraAt, &this->subCameraEye);
+    }
+}
+
+void BossGoma_Defeated(BossGoma* this, GlobalContext* globalCtx) {
+    static Vec3f _vel1 = { 0, 0, 0 };
+    static Vec3f _accel1 = { 0, 1, 0 };
+    static Color_RGBA8_n _color1 = { 255, 255, 255, 255 };
+    static Color_RGBA8_n _color2 = { 0, 100, 255, 255 };
+    static Vec3f _vel2 = { 0, 0, 0 };
+    static Vec3f _accel2 = { 0, -0.5f, 0 };
+    static Vec3f roomCenter = { -150, 0, -350 };
+
+    f32 dx;
+    f32 dz;
+    s16 j;
+    Vec3f vel1 = _vel1;
+    Vec3f accel1 = _accel1;
+    Color_RGBA8_n color1 = _color1;
+    Color_RGBA8_n color2 = _color2;
+    Vec3f vel2 = _vel2;
+    Vec3f accel2 = _accel2;
+    Vec3f pos;
+    Camera* camera;
+    Player* player = PLAYER;
```

Comment:
you should be able to just declare it non static and you can avoid making temps for them like this.
Id also avoid leading underscore names.
When you end up merging effects, `Color_RGBA8_n` will just be `Color_RGBA8`

Reply:
Ok, and I replaced `_vel` by `velInit` in `BossGoma_UpdateCeilingMovement` where I don't think I can get rid of static declarations?